Tests/DataLoaderTests/DataLoaderAbuseTests.swift

```diff
@@ -1,107 +1,114 @@
-import NIO
 import XCTest
 
 @testable import DataLoader
 
 /// Provides descriptive error messages for API abuse
 class DataLoaderAbuseTests: XCTestCase {
-    func testFuntionWithNoValues() throws {
-        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        defer {
-            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
-        }
-
+    func testFuntionWithNoValues() async throws {
         let identityLoader = DataLoader<Int, Int>(
             options: DataLoaderOptions(batchingEnabled: false)
         ) { _ in
-            eventLoopGroup.next().makeSucceededFuture([])
+            []
         }
 
-        let value = try identityLoader.load(key: 1, on: eventLoopGroup)
+        async let value = identityLoader.load(key: 1)
 
-        XCTAssertThrowsError(
-            try value.wait(),
-            "Did not return value for key: 1"
-        )
-    }
+        var didFailWithError: Error?
 
-    func testBatchFuntionMustPromiseAnArrayOfCorrectLength() throws {
-        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
-        defer {
-            XCTAssertNoThrow(try eventLoopGroup.syncShutdownGracefully())
+        do {
+            _ = try await value
+        } catch {
+            didFailWithError = error
         }
 
+        XCTAssertNotNil(didFailWithError)
```

Comment:
nit: Could we also check error messages like we did previously?

Reply:
Unfortunately, `XCTAssertNoThrow` does not work with asynchronous results. If you know a more elegant solution than the one I used, I would be happy to adopt it. Perhaps the new `Swift Testing` library supports working with asynchronous calls, but I haven't explored it yet.